### PR TITLE
frontend: refactor firmware-upgrade-required to functional component

### PR DESCRIPTION
**Pull request by thisconnect**
Changed FirmwareUpgradeRequired to use View component, also with this BitBox02.tsx does not need to use translation decorator anymore and types can be simplified.

Test by increasing lowestSupportedFirmwareVersion in:
- bitbox02-api-go/api/firmware/device.go

### CODE REVIEW

**Comment by thisconnect** [comment]
Screen Shot 2023-07-20 at 12 00 50

(new with this PR)

**Comment by thisconnect** [comment]
Screen Shot 2023-07-20 at 12 13 08

(current master)

**Comment by thisconnect** [comment]
This changed in the last release with the new settings revamp, as upgrade firmware component is shared here. Before it looked like this in v4.37.1, I guess this is what it was supposed to look like:

Screen Shot 2023-07-20 at 12 19 11

**Comment by thisconnect** [comment]
Added 2nd commit to restore old layout with just a button:

    
Screen Shot 2023-07-24 at 15 08 16

File: frontends/web/src/routes/settings/components/device-settings/firmware-setting.tsx [hunk 1]
```diff
@@ -55,14 +54,21 @@ const FirmwareSetting = ({ deviceID, versionInfo, asButton = false }: TProps) =>
...
-        className={`${asButton ? styles.asButton : ''}`}
-        settingName={t('deviceSettings.firmware.title')}
-        secondaryText={secondaryText}
-        onClick={canUpgrade ? () => setDialogOpen(true) : undefined}
-        displayedValue={versionInfo.currentVersion}
-        extraComponent={extraComponent}
-      />
+      { asButton ? (
+        <Button
+          onClick={canUpgrade ? () => setDialogOpen(true) : undefined}
```
Comment (shonsirsha) [inline frontends/web/src/routes/settings/components/device-settings/firmware-setting.tsx:59]: nit: we can make a small fn out of this for btn onClick and use this here and also at the other button.

**Review by shonsirsha — APPROVED** [review]
untested, code LGTM. Unfort cant really test it rn without a device.

File: frontends/web/src/routes/settings/components/device-settings/firmware-setting.tsx (hunk 1 above)
Comment (thisconnect) [reply frontends/web/src/routes/settings/components/device-settings/firmware-setting.tsx]: done

**Pull request merged**